pyccel/ast/variable.py

```diff
@@ -883,48 +901,40 @@ def __init__(
         rank  = base.rank
 
         # Add empty slices to fully index the object
-        if len(args) < rank:
-            args = args + tuple([Slice(None, None)]*(rank-len(args)))
+        if len(indices) < rank:
+            indices = indices + tuple([Slice(None, None)]*(rank-len(indices)))
 
-        if any(not isinstance(a, (int, TypedAstNode, Slice)) for a in args):
+        if any(not isinstance(a, (int, TypedAstNode, Slice)) for a in indices):
             errors.report("Index is not of valid type",
-                    symbol = args, severity = 'fatal')
+                    symbol = indices, severity = 'fatal')
 
-        self._indices = tuple(LiteralInteger(a) if isinstance(a, int) else a for a in args)
+        self._indices = tuple(LiteralInteger(a) if isinstance(a, int) else a for a in indices)
         super().__init__()
 
         # Calculate new shape
-
-        if shape is not None:
-            new_shape = []
-            from .mathext import MathCeil
-            for a,s in zip(args, shape):
-                if isinstance(a, Slice):
-                    start = a.start
-                    stop  = a.stop if a.stop is not None else s
-                    step  = a.step
-                    if isinstance(start, PyccelUnarySub):
-                        start = PyccelAdd(s, start, simplify=True)
-                    if isinstance(stop, PyccelUnarySub):
-                        stop = PyccelAdd(s, stop, simplify=True)
-
-                    _shape = stop if start is None else PyccelMinus(stop, start, simplify=True)
-                    if step is not None:
-                        if isinstance(step, PyccelUnarySub):
-                            start = s if a.start is None else start
-                            _shape = start if a.stop is None else PyccelMinus(start, stop, simplify=True)
-                            step = PyccelUnarySub(step)
-
-                        _shape = MathCeil(PyccelDiv(_shape, step, simplify=True))
-                    new_shape.append(_shape)
-            self._rank  = len(new_shape)
-            self._shape = None if self._rank == 0 else tuple(new_shape)
-        else:
-            new_rank = rank
-            for i in range(rank):
-                if not isinstance(args[i], Slice):
-                    new_rank -= 1
-            self._rank = new_rank
+        new_shape = []
+        from .mathext import MathCeil
+        for a,s in zip(indices, shape):
```

Comment:
How do we know that `shape is not None`?

Reply:
I spotted this thanks to the coverage results. The shape is set here:
https://github.com/pyccel/pyccel/blob/267be30007685fb23fa799be48c782fb144e46eb/pyccel/ast/variable.py#L198

And in the semantic stage an `IndexedElement` can only be called on an object with rank>1 (otherwise the Python code wouldn't be valid)